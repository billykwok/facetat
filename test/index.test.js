// @flow
import facetat from '../src';

import { css } from '@emotion/core';

const mq = facetat(
  { XS: 20, SM: 28, MD: 36, ML: 48, LG: 60, XL: 75 },
  { unit: 'rem' }
);

describe('Single-media form', () => {
  test('with tagged template literal', () => {
    const result = mq.XS`width: 18rem;`;
    expect(css(result)).toMatchSnapshot();
  });

  test('with multiple plain objects with a mixed of unit mode', () => {
    const result = mq.XS({ width: '240px' }, { fontSize: 16 });
    expect(css(result)).toMatchSnapshot();
  });

  test('with multiple plain objects with mixed units', () => {
    const result = mq.XS({ width: '240px' }, { fontSize: '1rem' });
    expect(css(result)).toMatchSnapshot();
  });

  test('with multiple plain objects', () => {
    const result = mq.XS({ width: 18 }, { fontSize: 16 });
    expect(css(result)).toMatchSnapshot();
  });

  test('with additional unitless height constraint', () => {
    const result = mq.XS.minHeight(40)({ width: 18 });
    expect(css(result)).toMatchSnapshot();
  });

  test('with additional height constraint', () => {
    const result = mq.XS.minHeight('640px')({ width: 18 });
    expect(css(result)).toMatchSnapshot();
  });
});

describe('Single-property form', () => {
  test('with number of argument more than that of breakpoints', () => {
    const result = mq.width('100%', 18, 24, 32, 44, 54, 72, 1000);
    expect(css(result)).toMatchSnapshot();
  });

  test('with number of argument fewer than that of breakpoints', () => {
    const result = mq.width('100%', 18, 24, 32, 44);
    expect(css(result)).toMatchSnapshot();
  });
});

describe('Chaining form', () => {
  test('with css', () => {
    const result = mq(
      css`
        width: 100%;
      `,
      css`
        width: 18rem;
      `
    );
    expect(css(result)).toMatchSnapshot();
  });

  test('with plain object', () => {
    const result = mq({ width: '100%' }, { width: '18rem' });
    expect(css(result)).toMatchSnapshot();
  });
});

describe('Compact form', () => {
  test('with plain object', () => {
    const result = mq({
      width: ['100%', 18, 24, 32, 44, 54, 72],
      background: ['red', null, 'blue', null, null, 'green']
    });
    expect(css(result)).toMatchSnapshot();
  });
});

describe('Misc usage', () => {
  test('with customized unitless min-width', () => {
    const result = mq.minWidth(40)({ width: '100%' });
    expect(css(result)).toMatchSnapshot();
  });

  test('with customized min-width', () => {
    const result = mq.minWidth('640px')({ width: '100%' });
    expect(css(result)).toMatchSnapshot();
  });
  test('with customized unitless min-height', () => {
    const result = mq.minHeight(40)({ width: '100%' });
    expect(css(result)).toMatchSnapshot();
  });

  test('with customized min-height', () => {
    const result = mq.minHeight('640px')({ width: '100%' });
    expect(css(result)).toMatchSnapshot();
  });
});
