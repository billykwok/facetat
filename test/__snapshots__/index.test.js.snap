// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chaining form with css 1`] = `
Object {
  "map": undefined,
  "name": "17d7cf0",
  "next": undefined,
  "styles": "
        width: 18rem;
      ",
}
`;

exports[`Chaining form with plain object 1`] = `
Object {
  "map": undefined,
  "name": "m3h2rj",
  "next": undefined,
  "styles": "width:18rem;",
}
`;

exports[`Compact form with plain object 1`] = `
Object {
  "map": undefined,
  "name": "1mcmhjb",
  "next": undefined,
  "styles": "width:100%;background:red;@media(min-width:20rem){width:18rem;}@media(min-width:28rem){width:24rem;background:blue;}@media(min-width:36rem){width:32rem;}@media(min-width:48rem){width:44rem;}@media(min-width:60rem){width:54rem;background:green;}@media(min-width:75rem){width:72rem;}",
}
`;

exports[`Misc usage with customized min-height 1`] = `
Object {
  "map": undefined,
  "name": "1i86alf",
  "next": undefined,
  "styles": "@media(min-height:640px){width:100%;}",
}
`;

exports[`Misc usage with customized min-width 1`] = `
Object {
  "map": undefined,
  "name": "ccf86",
  "next": undefined,
  "styles": "@media(min-width:640px){width:100%;}",
}
`;

exports[`Misc usage with customized unitless min-height 1`] = `
Object {
  "map": undefined,
  "name": "4qmkok",
  "next": undefined,
  "styles": "@media(min-height:40rem){width:100%;}",
}
`;

exports[`Misc usage with customized unitless min-width 1`] = `
Object {
  "map": undefined,
  "name": "sk8x50",
  "next": undefined,
  "styles": "@media(min-width:40rem){width:100%;}",
}
`;

exports[`Single-media form with additional height constraint 1`] = `
Object {
  "map": undefined,
  "name": "ztzb64",
  "next": undefined,
  "styles": "@media(min-width:20rem) and (min-height:640px){width:18px;}",
}
`;

exports[`Single-media form with additional unitless height constraint 1`] = `
Object {
  "map": undefined,
  "name": "1ktezg3",
  "next": undefined,
  "styles": "@media(min-width:20rem) and (min-height:40rem){width:18px;}",
}
`;

exports[`Single-media form with multiple plain objects 1`] = `
Object {
  "map": undefined,
  "name": "18t2b45",
  "next": undefined,
  "styles": "@media(min-width:20rem){width:18px;font-size:16px;}",
}
`;

exports[`Single-media form with multiple plain objects with a mixed of unit mode 1`] = `
Object {
  "map": undefined,
  "name": "199dyzj",
  "next": undefined,
  "styles": "@media(min-width:20rem){width:240px;font-size:16px;}",
}
`;

exports[`Single-media form with multiple plain objects with mixed units 1`] = `
Object {
  "map": undefined,
  "name": "zcnrwc",
  "next": undefined,
  "styles": "@media(min-width:20rem){width:240px;font-size:1rem;}",
}
`;

exports[`Single-media form with tagged template literal 1`] = `
Object {
  "map": undefined,
  "name": "et70wo",
  "next": undefined,
  "styles": "@media(min-width:20rem){width: 18rem;}",
}
`;

exports[`Single-property form with number of argument fewer than that of breakpoints 1`] = `
Object {
  "map": undefined,
  "name": "1qtyuct",
  "next": undefined,
  "styles": "width:18rem;@media(min-width:28rem){width:24rem;}@media(min-width:36rem){width:32rem;}@media(min-width:48rem){width:44rem;}",
}
`;

exports[`Single-property form with number of argument more than that of breakpoints 1`] = `
Object {
  "map": undefined,
  "name": "1pi0qkl",
  "next": undefined,
  "styles": "width:18rem;@media(min-width:28rem){width:24rem;}@media(min-width:36rem){width:32rem;}@media(min-width:48rem){width:44rem;}@media(min-width:60rem){width:54rem;}@media(min-width:75rem){width:72rem;}",
}
`;
